api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -160,6 +169,27 @@ private void handleValueCodedConcept(Criteria criteria, TokenAndListParam valueC
 		}
 	}
 	
+	private Boolean validReferenceParam(ReferenceParam ref) {
+		return (ref != null && ref.getIdPart() != null && ref.getResourceType() != null);
+	}
+	
+	private void handleReference(Criteria criteria, ReferenceAndListParam reference, String property, String alias) {
+		handleAndListParam(reference, param -> {
+			if (validReferenceParam(param)) {
+				if (lacksAlias(criteria, alias)) {
+					criteria.createAlias(property, alias);
+				}
+				
+				List<Optional<Criterion>> criterionList = new ArrayList<>();
+				criterionList.add(Optional.of(eq(String.format("%s.reference", alias), param.getIdPart())));
+				criterionList.add(Optional.of(eq(String.format("%s.type", alias), param.getResourceType())));
+				return Optional.of(and(toCriteriaArray(criterionList)));
```

Comment:
Do we actually need this?

Reply:
Also, note that these are being translated into an actual Hibernate query; is Hibernate going to know how to handle `basedOnReference.reference = <id part>`?